src/sentry/models/user.py

```diff
@@ -78,7 +80,13 @@ def delete(self):
     def save(self, *args, **kwargs):
         if not self.username:
             self.username = self.email
-        return super(User, self).save(*args, **kwargs)
+        user = super(User, self).save(*args, **kwargs)
+        try:
```

Comment:
this would be better to do with a post_save signal that checks "Was User created or just changed" so we dont have to force this extra query on every user.save() call

Take a look at sentry.receivers for some examples